modules/apps/callbacks/testing/simapp/contract_keeper.go

```diff
@@ -145,16 +162,25 @@ func (k ContractKeeper) processMockCallback(
 	// increment callback execution attempts
 	k.Counters[callbackType]++
 
-	if gasRemaining < 500000 {
-		// consume gas will panic since we attempt to consume 500_000 gas, for tests
-		ctx.GasMeter().ConsumeGas(500000, fmt.Sprintf("mock %s callback panic", callbackType))
-	}
-
-	if authAddress == MockCallbackUnauthorizedAddress {
-		ctx.GasMeter().ConsumeGas(500000, fmt.Sprintf("mock %s callback unauthorized", callbackType))
+	switch contractAddress {
+	case ErrorContract:
+		ctx.GasMeter().ConsumeGas(gasRemaining/2, fmt.Sprintf("mock %s callback unauthorized", callbackType))
```

Comment:
How come it needs to be gas remaining / 2? or does this have any implications?

Reply:
I just wanted an amount that wouldn't panic

Reply:
should we add this reasoning in the godoc?